# Bonus: Temperature Analysis I

In [28]:
import pandas as pd
from datetime import datetime as dt

In [40]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [41]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])
df.head()


station       date  prcp  tobs
0  USC00519397 2010-01-01  0.08    65
1  USC00519397 2010-01-02  0.00    63
2  USC00519397 2010-01-03  0.00    74
3  USC00519397 2010-01-04  0.00    76
4  USC00519397 2010-01-06   NaN    73

In [31]:
# Set the date column as the DataFrame index
# df = df.set_index('date')

In [42]:
# Drop the date column
df.head()

station       date  prcp  tobs
0  USC00519397 2010-01-01  0.08    65
1  USC00519397 2010-01-02  0.00    63
2  USC00519397 2010-01-03  0.00    74
3  USC00519397 2010-01-04  0.00    76
4  USC00519397 2010-01-06   NaN    73

### Compare June and December data across all years 

In [46]:
from scipy import stats
import numpy as np

In [44]:
# Filter data for desired months
June = df[df['date'].dt.month == 6]
Dec = df[df['date'].dt.month == 12]
print(Dec)


           station       date  prcp  tobs
305    USC00519397 2010-12-01  0.04    76
306    USC00519397 2010-12-03  0.00    74
307    USC00519397 2010-12-04  0.00    74
308    USC00519397 2010-12-06  0.00    64
309    USC00519397 2010-12-07  0.00    64
...            ...        ...   ...   ...
19323  USC00516128 2016-12-27  0.14    71
19324  USC00516128 2016-12-28  0.14    71
19325  USC00516128 2016-12-29  1.03    69
19326  USC00516128 2016-12-30  2.37    65
19327  USC00516128 2016-12-31  0.90    65

[1517 rows x 4 columns]


In [47]:
# Identify the average temperature for June
avgJune = np.mean(June['tobs'])
print(avgJune)

74.94411764705882


In [48]:
# Identify the average temperature for December
avgDec = np.mean(Dec['tobs'])
print(avgDec)

71.04152933421226


In [ ]:
# Create collections of temperature data
a = June['tobs'].to_numpy()
b = Dec['tobs'].to_numpy()

In [52]:
# Run unpaired t-test (independent)
## compare the means of two groups - are they equal or significantly different?
## Null hypothesis: mean temperatures are the same
## Alternative hypothesis: June is warmer than December

stats.ttest_ind(a=a, b=b, equal_var=True)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

We ran an unpaired t-test because we are comparing two populations (December temperatures and June temperatures). Our results are a t-test score of 31.60372399000329, and a p-value of 3.9025129038616655e-191.

Our null hypothesis is that the mean temperatures between June and December are the same. The p-value is lower than the 5% threshold, showing that June temperatures are warmer than December.